features/step_definitions/dish_steps.rb

```diff
@@ -0,0 +1,9 @@
+Given(/^the following dishes exist$/) do |table|
+  table.hashes.each do |hash|
+    rest = Restaurant.find_by(name: hash["restaurant"])
+    hash["menu"] = Menu.where(name: hash["menu"], restaurant_id: rest.id).first
+    hash["price"] = hash["price"].to_i
+    restaurant = hash.delete("restaurant")
+    FactoryGirl.create(:dish, hash)
+  end
```

Comment:
Line 6 now reads `hash.delete("restaurant")`

Line 5. Price value in the hash is converted to an integer as the dish model requires the price to be an integer

Line 6. Restauraunt key value is deleted form the hash as the restaurant is only used to make sure that we're getting the correct menu, as searched for on Line 4 in the code (several restaurants could have a breakfast menu). After we've got the correct menu we need to delete the restaurant from the hash before we can call Factory Girl.

Line 7. Factory Girl creates a Dish from the values given in the hash